pkg/controller/factory/factory.go

```diff
@@ -54,6 +77,26 @@ func (f *Factory) WithInformers(informers ...Informer) *Factory {
 	return f
 }
 
+// WithInformerQueueKeys allows to pass multiple InformerQueueKeyFunc that return informer and the corresponding queue key function
+// that will be used for the particular informer.
+func (f *Factory) WithInformerQueueKeyFuncs(mappings ...InformerQueueKeyFunc) *Factory {
```

Comment:
this looks like Scala

Reply:
Why no simple `WithInformers(keyFunc, informers....)`? If you want different keyFuncs, call it multiple times. The functional informer type works, but feels misplaced and oversized.

Reply:
neat, right?

Reply:
Scala does not mean "neat". It means "what the hell are you doing here?" ;-)